test/serializers/attribute_test.rb

```diff
@@ -71,6 +71,21 @@ def id
 
         assert_equal('custom', hash[:blog][:id])
       end
+
+      PostWithVirtualAttribute = Class.new(::Model)
+      class PostWithVirtualAttributeSerializer < ActiveModel::Serializer
+        attribute :name do
+          "#{object.first_name} #{object.last_name}"
+        end
+      end
```

Comment:
I think that the problem here is that the key may not be the name of the attribute that we want to read, as in:

```
class MyThingSerializer < ActiveModel::Serializer
  attribute :name, key: :object
end
```

This should render something like `{ "object": "my name" }`, but to get that, we actually need to ~~invoke the `name` method on the object~~ read the `name` attribute for serialization.